### PR TITLE
Update REGEX

### LINKED ISSUE #101: Undo compatibility with Archive/Load

### LINKED ISSUE #134: Update Commands in AutoComplete
Add the following into AutoComplete:
- Load
- Filter

### LINKED ISSUE #135: Add REGEX constraints for Person's fields
Limit the characters of the following fields:
- Phone to 8.
- Module to 30.
- Name to 100.
- Tag to 30.

Update REGEX for the following fields:
- Phone (number only)
- Module(alphanumeric)
- Grade (0-100)
- Name (alphabet, hyphen, dot, comma)

### LINKED ISSUE #139: Update UI for Grade

### LINKED ISSUE #145: Update UG with Undo and Redo command

### PR DESCRIPTION
**Pull request by RadieonAjax**
### What's Changed

- Enhanced input validation for Person subclasses by updating regular expressions (REGEX) for each field to enforce stricter formatting rules.
- Added length checks on inputs to ensure compliance with field constraints.
- Added Filter and Load to AutoComplete.
- Changed how grades are shown to user - User need to hover over the module to view the grade.
   -  Grades over 50 will display green, while grades lesser than 50 will display red. Ungraded modules will be greyed out.

### Bug Fixes

- Resolved an issue where Undo and Redo commands incorrectly applied after executing Archive or Load commands.

Closes #101.
Closes #134.
Closes #135.
Closes #139.
Closes #145.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ArchiveCommand.java [hunk 1]
```diff
@@ -34,6 +34,7 @@ public Path getArchivePath() {
     public CommandResult execute(Model model) {
         requireNonNull(model);
         model.setAddressBook(new AddressBook());
```
Comment (Jianwen0451) [inline src/main/java/seedu/address/logic/commands/ArchiveCommand.java:36]: This line is originally act as clear, if you are sure that your clear command is working correctly, you can remove this line

**Review by Jianwen0451 — COMMENTED** [review]
I actually do not really understand why the test fails, but I think should make it pass before we merge

File: src/main/java/seedu/address/logic/commands/ArchiveCommand.java (hunk 1 above)
Comment (RadieonAjax) [reply src/main/java/seedu/address/logic/commands/ArchiveCommand.java:36]: The current implementation for setAddressBook doesn't replace the current model with a new model, but rather, it updates the Persons List with what you put in here.

In this case, it resets the Persons List, but the AddressBookState List is still not cleared, hence necessitating the clearAddressBook.

**Comment by RadieonAjax** [comment]
> I actually do not really understand why the test fails, but I think should make it pass before we merge

I've found the issue and pushed the update!

**Review by yongqqqq — APPROVED** [review]
Overall, good job! I am looking forward to seeing the REGEX in action!

File: docs/UserGuide.md [hunk 2]
```diff
@@ -145,6 +145,29 @@ Examples:
...
+Restores the last action that was undone, allowing you to recover data after an undo operation.
+
+Format: `redo`
+
+* **Note:** This command does not work with the `list`, `filter`, or `find` commands.
+
+Examples:
+* `redo` will reapply the last command that was undone, restoring the previous state of the address book.
+
+
```
Comment (yongqqqq) [inline docs/UserGuide.md:170]: I think you can include screenshots to better showcase the changes

**Review by yongqqqq — APPROVED** [review]
Overall, good job! LGTM!

**Review by Jianwen0451 — APPROVED** [review]
LGTM!

**Pull request merged**